com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -142,38 +347,129 @@ public override void OnNetworkSpawn()
+            m_ClientSendList.Clear();
+            m_ClientSendList.Add(playerId);
+            m_ClientRpcParams.Send.TargetClientIds = m_ClientSendList;
+            // With synchronization we send all parameters
+            m_ParametersToUpdate.Clear();
+            for (int i = 0; i < m_CachedAnimatorParameters.Length; i++)
+            {
+                m_ParametersToUpdate.Add(i);
+            }
+            SendParametersUpdate(m_ClientRpcParams);
+            for (int layer = 0; layer < m_Animator.layerCount; layer++)
+            {
+                AnimatorStateInfo st = m_Animator.GetCurrentAnimatorStateInfo(layer);
+
+                var stateHash = st.fullPathHash;
+                var normalizedTime = st.normalizedTime;
+                var totalSpeed = st.speed * st.speedMultiplier;
+                var adjustedNormalizedMaxTime = totalSpeed > 0.0f ? 1.0f / totalSpeed : 0.0f;
+                // NOTE:
+                // When synchronizing, for now we will just complete the transition and
+                // synchronize the player to the next state being transitioned into
+                if (m_Animator.IsInTransition(layer))
+                {
+                    var tt = m_Animator.GetAnimatorTransitionInfo(layer);
+                    var nextState = m_Animator.GetNextAnimatorStateInfo(layer);
+
+                    if (nextState.length > 0)
+                    {
+                        var nextStateTotalSpeed = nextState.speed * nextState.speedMultiplier;
+                        var nextStateAdjustedLength = nextState.length * nextStateTotalSpeed;
+                        // TODO: We need to get the transition curve for the target state as well as some
+                        // reasonable RTT estimate in order to get a more precise normalized synchronization time
+                        var transitionTime = Mathf.Min(tt.duration, tt.duration * tt.normalizedTime) * 0.5f;
+                        normalizedTime = Mathf.Min(1.0f, transitionTime > 0.0f ? transitionTime / nextStateAdjustedLength : 0.0f);
+                    }
+                    else
+                    {
+                        normalizedTime = 0.0f;
+                    }
+
+                    stateHash = nextState.fullPathHash;
+                }
+                else
+                if (st.normalizedTime >= adjustedNormalizedMaxTime)
+                {
+                    continue;
+                }
+
+                var animMsg = new AnimationMessage
+                {
+                    StateHash = stateHash,
+                    NormalizedTime = normalizedTime,
+                    Layer = layer,
+                    Weight = m_LayerWeights[layer]
+                };
+                // Server always send via client RPC
+                SendAnimStateClientRpc(animMsg, m_ClientRpcParams);
+            }
+        }
+
+        private void Server_OnClientConnectedCallback(ulong playerId)
```

Comment:
```suggestion
        private void ServerOnClientConnectedCallback(ulong playerId)
```

Reply:
we shouldn't use underscore character here, maybe rename the method as something else if `ServerOnClient...` doesn't look good to you?

Reply:
I can just rename it to OnClientConnectedCallback...

Reply:
Applied alternate to suggestion, but accomplishes the same goal.